grpc-ballerina/tests/28_unary_basic_auth_service.bal

```diff
@@ -25,11 +26,19 @@ service "HelloWorld28" on ep28 {
     isolated remote function testStringValueReturn(ContextString request) returns string|error {
         if !request.headers.hasKey(AUTH_HEADER) {
             return error AbortedError("AUTH_HEADER header is missing");
-        } else if request.headers.get(AUTH_HEADER) == "Basic YWRtaW46MTIz" {
-            return "Hello WSO2";
         } else {
-            return error UnauthenticatedError(string `Invalid basic auth token: ${<string>request.headers.get(
-            AUTH_HEADER)}`);
+            ListenerFileUserStoreBasicAuthHandler handler = new;
+            auth:UserDetails|UnauthenticatedError authResult = handler.authenticate(request.headers);
+            if (authResult is UnauthenticatedError) {
+                return authResult;
+            } else {
+                PermissionDeniedError? authrzResult = handler.authorize(<auth:UserDetails>authResult, "write");
+                if (authrzResult is ()) {
+                    return "Hello WSO2";
+                } else {
+                    return authrzResult;
+                }
+            }
```

Comment:
Just a suggestion. In case you need to distinguish the variables for authentication and authorization, we can follow the `authn` and `authz` names.
```suggestion
            auth:UserDetails|UnauthenticatedError authnResult = handler.authenticate(request.headers);
            if (authnResult is UnauthenticatedError) {
                return authnResult;
            } else {
                PermissionDeniedError? authzResult = handler.authorize(<auth:UserDetails>authnResult, "write");
                if (authzResult is ()) {
                    return "Hello WSO2";
                } else {
                    return authzResult;
                }
            }
```